// @ionic/angular/esm5/providers/gesture-controller.js
import * as tslib_1 from "tslib";
import { Injectable } from '@angular/core';
import { createGesture } from '@ionic/core';
import * as i0 from "@angular/core";
import * as ɵngcc0 from '@angular/core';
var GestureController = /** @class */ (function () {
    function GestureController() {
    }
    /**
     * Create a new gesture
     */
    GestureController.prototype.create = function (opts) {
        return createGesture(opts);
    };
    GestureController.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function GestureController_Factory() { return new GestureController(); }, token: GestureController, providedIn: "root" });
GestureController.ɵfac = function GestureController_Factory(t) { return new (t || GestureController)(); };
GestureController.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: GestureController, factory: function (t) { return GestureController.ɵfac(t); }, providedIn: 'root' });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(GestureController, [{
        type: Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return []; }, null); })();
    return GestureController;
}());
export { GestureController };
